src/hooks/usePushNotifications.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { useHttpClient } from './useHttpClient';
+import { Account } from './useAccounts';
+import { useActiveAccount } from './useActiveAccount';
+import {
+  getInitialNotification,
+  onNotificationOpened,
+  onNotificationReceived,
+} from 'src/common';
+import { Platform } from 'react-native';
+import { Notifications } from 'react-native-notifications';
+import { AxiosInstance } from 'axios';
+
+interface PushNotificationsStateType {
+  events: Event | { type: string; notification: Notification }[];
+  setEvents: React.Dispatch<
+    React.SetStateAction<Event | { type: string; notification: Notification }[]>
+  >;
+  httpClient: AxiosInstance;
+  account: Account | undefined;
+}
+
+export const PushNotificationsContext = createContext<
+  PushNotificationsStateType | undefined
+>(undefined);
+
+export function PushNotificationsProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [events, setEvents] = useState<
+    Event | { type: string; notification: Notification }[]
+  >([]);
+  const { httpClient } = useHttpClient();
+  const { account } = useActiveAccount();
+
+  useEffect(() => {
+    // Handler called when a notification is pressed
+    onNotificationOpened((notification) => {
+      setEvents((events) => [
```

Comment:
typescript was complaining about typing here, so i tried strongly typing `events` and `setEvents` which caused a cascade of type errors. So to resolve had to do the ugly things called out elsewhere in this PR. I guess I couldn't quite figure out the right typing